aii-opennebula/src/main/perl/opennebula.pm

```diff
@@ -86,9 +86,14 @@ sub make_one
 # Detect and process ONE templates
 sub process_template
 {
-    my ($self, $config, $tt_name) = @_;
+    my ($self, $config, $tt_name, $oneversion) = @_;
     
     my $tree = $config->getElement('/')->getTree();
+    if ((defined $oneversion) and ($oneversion >= version->new("4.9.0"))) {
+        $tree->{system}->{opennebula}->{boot} = $oneversion;
```

Comment:
I was also thinking about this. Probably in the future more things will change with another update in a different section. Maybe we can create a new function called `update_config_tree` or something like that, this function could update, depending on the version detected by the RPC, the conf tree, so we will set within this method `$tree->{system}->{opennebula}->{$section}=$version` if it's required and return the modified tree. After these changes we only use the TT file to write a specific conf for a specific version, if the version is not set we write the default setup, instead to include all the logic in the TT file. 

What do you think? I'm not sure what is the best way to handle this...